File: api/app/db/models/todo.py
from datetime import datetime, timezone
from uuid import uuid4

from sqlalchemy import Boolean, DateTime, String
from sqlalchemy.orm import Mapped, mapped_column

from app.db.base import Base


def utc_now() -> datetime:
    return datetime.now(timezone.utc)


class Todo(Base):
    __tablename__ = "todos"

    id: Mapped[str] = mapped_column(String(36), primary_key=True, default=lambda: str(uuid4()))
    description: Mapped[str] = mapped_column(String(500), nullable=False)
    completed: Mapped[bool] = mapped_column(Boolean, default=False, nullable=False)
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), default=utc_now, nullable=False)
    updated_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), default=utc_now, onupdate=utc_now, nullable=False)
    user_id: Mapped[str | None] = mapped_column(String(36), nullable=True)

File: web/src/features/todos/types/todo.ts
export type Todo = {
  id: string
  description: string
  completed: boolean
  created_at: string
  updated_at: string
}

export type TodoListResponse = {
  data: Todo[]
}

export type TodoResponse = {
  data: Todo
}

export type CreateTodoInput = {
  description: string
}

export type UpdateTodoInput = {
  description?: string
  completed?: boolean
}
